### PR TITLE
Update routes on yeoman code generation

### PR DESCRIPTION
**Pull request by pbarrios**
## Background

Hi guys, I'm still working in the complete automation of the code generation for the crud operations. This time I added the update of the routes. 

## Changes done

- Minor change in the routes.js file.
- Add method to generate the new routes based on a template.
- Update the readme file.

### CODE REVIEW

File: app/routes.js [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
-    <Route path="/app/users/:id/edit" component={UserEditPage} />
-    <Route path="/app/about" component={AboutPage} />
-    <Route path="*" component={NotFound} />
-  </Route>
+const defaultExport = (
+    <Route path='/app' component={App}>
+      <Redirect from='/' to='/app'/>
+      <IndexRoute component={HomePage}/>
+      <Route path='/app/about' component={AboutPage}/>
+      <Route path='*' component={NotFound}/>
```
Comment (pbarrios) [inline app/routes.js]: @pablomigliasso note that I have moved this route, does the order matter? If so, I have to change the code generation.

File: cli/generators/crud/crudGenerator.js [hunk 2]
```diff
@@ -57,4 +57,43 @@ module.exports = class extends Generator {
...
+            var currentImports = tree.body.node;
+
+            var renderedTemplateTree = program(renderedTemplate, null, {sourceType: 'module', plugins: { jsx: true } });
+            var newRoutes = renderedTemplateTree.var('defaultExport').value().children;
+            var newImports = renderedTemplateTree.body.node.filter(function(node) {
+              return node.type === 'ImportDeclaration';
+            });
+
+            newImports.forEach(function(newImport) {
+              currentImports.splice(4, 0, newImport);
```
Comment (AlphaGit) [inline cli/generators/crud/crudGenerator.js]: What is this instruction meaning to do? I couldn't quite figure it out.

File: cli/generators/crud/crudGenerator.js [hunk 3]
```diff
@@ -57,4 +57,43 @@ module.exports = class extends Generator {
...
+            
+            var insertRouteAt = currentRoutes.findIndex(function(route){
+              return route.openingElement && route.openingElement.name.name === "IndexRoute";
+            });
+
+            newRoutes.forEach(function(newRoute) {
+                currentRoutes.splice(++insertRouteAt, 0, newRoute);
+            });
+
+            return tree.toString();
```
Comment (AlphaGit) [inline cli/generators/crud/crudGenerator.js]: Is modifying the `currentRoutes` variable enough? Don't you need to re-add it to the tree?

File: cli/generators/crud/crudGenerator.js (hunk 2 above)
Comment (pbarrios) [inline cli/generators/crud/crudGenerator.js]: It's for insert after the 4th current import. But now that you mention, I can add it at the beginning. I will, change it, thanks!

File: cli/generators/crud/crudGenerator.js (hunk 3 above)
Comment (pbarrios) [inline cli/generators/crud/crudGenerator.js]: Yes, by modifying the `currentRoutes` variable is enough.

**Review by mravinale — APPROVED** [review]

**Review by AlphaGit — APPROVED** [review]
@pbarrios Thanks!

**Pull request closed without merging**